app/Application/Services/WalletBalanceService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Application\Services;
+
+use App\Application\Exceptions\WalletNotFoundException;
+use App\Domain\DataSources\CoinDataSource;
+use App\Domain\DataSources\WalletDataSource;
+use Illuminate\Support\Facades\Cache;
+
+class WalletBalanceService
+{
+    private WalletDataSource $walletDataSource;
+    private CoinDataSource $coinDataSource;
+    public function __construct(WalletDataSource $walletDataSource, CoinDataSource $coinDataSource)
+    {
+        $this->walletDataSource = $walletDataSource;
+        $this->coinDataSource = $coinDataSource;
+    }
+
+    /**
+     * @throws WalletNotFoundException
+     */
+    public function getsBalance($walletId)
+    {
+        if (is_null($this->walletDataSource->findById($walletId))) {
```

Comment:
Here we have to be careful.
We are calling the walletDataSource->findById this function what it does is to return us a wallet|null and then what we do is to call to a function so that it returns us another wallet?

Reply:
Maybe it´s better findById return us a bool? Or if u want to findById returns a wallet you don´t need getWalletById